src/testUtils/index.tsx

```diff
@@ -15,31 +47,79 @@ export interface QueryString {
     [key: string]: string;
 }
 
-// To-do: improve this API.
-// shallowRender should default to true,
-// shouldn't have to pass an empty propOverrides object to get a shallow-rendered wrapper with no prop overrides, e.g.:
-// const { wrapper } = setup({}, true);
+export const contextTypes: { [key: string]: Function } = {
+    store: object,
+    endpoints: endpointsPT,
+    languages: languagesPT,
+    flow: flowPT
+};
+
+export const baseState: AppState = {
+    ...initialState,
+    ...{
+        flowContext: {
+            ...initialState.flowContext,
+            definition: require('../../__test__/flows/colors.json') as FlowDefinition
+        }
+    },
+    ...{
+        flowEditor: {
+            ...initialState.flowEditor,
+            editorUI: {
+                ...initialState.flowEditor.editorUI,
+                language: getBaseLanguage((config as FlowEditorConfig).languages)
+            }
+        }
+    }
+};
+
+export const configProviderContext: ConfigProviderContext = {
+    endpoints: (config as FlowEditorConfig).endpoints,
+    languages: (config as FlowEditorConfig).languages,
+    flow: (config as FlowEditorConfig).flow
+};
+
+export const setMock = (implementation?: (...args: any[]) => any): Query<jest.Mock> =>
+    set(jest.fn(implementation));
+
 /**
  * Compose setup method for component tests
  */
-export const createSetup = <P extends {}, C extends ConfigProviderContext = ConfigProviderContext>(
+export const composeSetup = <P extends {}>(
```

Comment:
finally updating this bad boy. modified to allow us to update `store` state, utilize `immutability-helper`. 'compose' because it's a higher-order function.

Reply:
forgot to mention it also passes context, store by default so we don't have to initialize them in each test